backend/controllers/Events.js

```diff
@@ -0,0 +1,74 @@
+const express = require('express')
+const bodyParser = require('body-parser')
+const cors = require('cors')
+const eventRouter = express.Router()
+
+var mongodb = require('mongodb');
+var Event = require('../models/Event');
+
+eventRouter.use(cors())
+eventRouter.use(bodyParser.json());
+eventRouter.use(bodyParser.urlencoded({
+  extended: true
+}));
+
+eventRouter.get('/list', function(req, res) {
+  Event.find(function(err, events) {
+		if (err) {
+			res.json(err);
+		}
+		events = events.map(q => q.toJSONFor());
+		res.json(events);
+	});
+})
+
+eventRouter.delete('/delete', function(req, res) {
+	var ids = req.body.eventIDs.map(function(id) {
+		return new mongodb.ObjectID(id);
+	});
+
+	Event.deleteMany({_id: {$in: ids}}, function(err) {
+		if (err) {
+			res.send(err);
+		}
+  });
+	res.send({"message": "Deleted event!"});
+})
+
+eventRouter.post('/insert', function(req, res) {
+ 	var event = new Event();
+ 	event.title = req.body.title;
+  event.description = req.body.description;
+ 	event.startTime = req.body.startTime;
+ 	event.endTime = req.body.endTime;
+ 	event.location = req.body.location;
+ 	event.grade = req.body.grade;
+  event.imageURI = req.body.imageURI;
+
+ 	event.save(function(err) {
+ 		res.json({
+ 		  message: 'Event created!',
+ 	 	  event: event
+   });
+ 	});
+ });
+
+ eventRouter.post('/edit', function(req, res) {
+	var eventToEdit = req.body;
+
+	Event.findById(eventToEdit.id, function(err, event) {
+		event.set(eventToEdit);
```

Comment:
Do we require event validation? What kind of schema is events sent in?

Reply:
Everything is a string except for grade (number) and the start/end times (date). I think it might make more sense to do the validation on the client side? We can check for valid grades/times (not sure if this is already implemented).

Reply:
Yeah having client side checks is probably easier. I just don't want a case where someone is testing something and it breaks the db.

cc: @theRoughCode Any thoughts on preferring backend vs frontend validation?

Reply:
Yea, I think it'll be easier to do it in the front-end.  If it comes up that it requires more work, we can shift it to the back-end in the future.